src/ray/raylet/node_manager.cc

```diff
@@ -517,7 +517,9 @@ void NodeManager::NodeAdded(const GcsNodeInfo &node_info) {
   if (node_id == self_node_id_) {
     // We got a notification for ourselves, so we are connected to the GCS now.
     // Save this NodeManager's resource information in the cluster resource map.
-    cluster_resource_map_[node_id] = initial_config_.resource_config;
+    if (0 == cluster_resource_map_.count(node_id)) {
```

Comment:
why need this if?

Reply:
This if is used to filter duplicate data, and i have moved it to the beginning of the function.